### DISCUSSION TITLE
Any insight into why the command npm install is taking hours and hours to complete?

### DISCUSSION THREAD

**Discussion by paparush** [pnp/sp-dev-fx-webparts#3539] [General]
Cloned the repo.  Inside of VS Code I navigate the terminal into /sp-dev-fx-webparts/tutorials/tutorial-getting-started/helloworld-webpart and issue the command: npm install.

4 hours later I'm still watching npm messages slowly accumulate in the terminal window.

**Comment by Link631** [comment]
Its due to the gulp packaging.

For me I solved it as followed:

1. dev install spfx-fast-serve (https://www.npmjs.com/package/spfx-fast-serve)
- it gives a huge performanceboost in hot deploy while developing and also a little boost for packaging

2. run node with npm in linux subsystem for windows (wsl2)

regards.

**Comment by Fringie** [comment]
There are like 100+ projects in this repo. I've never tried to install all the projects at once, but I imagine it'd take hours to do which is why you're running into slow performance - projects usually take like 3-5 minutes each to install for me, now multipfy that by 100+. 

I normally get the solution I'm interested in then just install that one.
